web/src/compose_banner.ts

```diff
@@ -66,14 +66,14 @@ export function get_compose_banner_container($textarea: JQuery): JQuery {
 // to a banner container. The function accepts a container element
 // as a parameter, to which a banner should be appended.
 export function append_compose_banner_to_banner_list(
-    banner: HTMLElement | JQuery.htmlString,
+    banner: JQuery | JQuery.htmlString,
```

Comment:
Oh wow this unholy union shouldn’t have been allowed. Interpreting a string as HTML is dangerous and should always be an explicit operation. Let’s restrict to `$banner: JQuery`, and fix any callers that want to pass an HTML string to convert it to `JQuery` with `$(str)` first.